src/tasks/taskset.ts

```diff
@@ -1,12 +1,21 @@
-import { SingleTask } from "./taskinterfaces";
+import { IAction } from "../classes";
+import { TaskBase } from "./taskinterfaces";
 
 /** @hidden */
-export class TaskSet {
+export class TaskSet implements TaskBase {
+
     constructor(
         public readonly isCompleted: boolean,
         public readonly isFaulted: boolean,
-        public readonly tasks: SingleTask[],
+        private readonly tasks: TaskBase[],
+        private readonly completionIndex?: number,
         public result?: unknown,
         public exception?: Error,
     ) { }
+
+    public yieldNewActions(): IAction[] {
+        // Get all of the actions in subtasks and flatten into one array.
+        return this.tasks.map((task) => task.yieldNewActions())
+            .reduce((acc, arr) => acc.concat(arr));
```

Comment:
I can't tell what type of objects `acc` and `arr` are. Can we use more descriptive variable names for those?